python/tank/descriptor/io_descriptor/base.py

```diff
@@ -627,6 +628,12 @@ def _get_cache_paths(self):
         paths.append(self._get_primary_cache_path())
         return paths
 
+    def _get_temporary_cache_path(self):
+        """
+        Returns the temporary download cache path for this descriptor.
+        """
+        return os.path.join(self._bundle_cache_root, "tmp", uuid.uuid4().hex)
```

Comment:
is this needed on the base class level? why not keep it on the `IODescriptorDownloadable` level?